src/views/design/ScriptDetail/EditAction/index.tsx

```diff
@@ -143,22 +153,72 @@ function EditAction({
                     className={classnames(classes.actionName)}
                     width="60%"
                 >
-                    <TextField
-                        id="action-name"
-                        label={translator('scripts.detail.edit_action.name')}
-                        defaultValue={name}
-                        onBlur={(e) => setName(e.target.value)}
-                        className={classes.nameTextField}
-                        InputProps={{
-                            readOnly: !isCreateScriptRoute
-                                && !checkAuthority(SECURITY_PRIVILEGES.S_SCRIPTS_WRITE, securityGroupName),
-                            disableUnderline: !isCreateScriptRoute
-                                && !checkAuthority(SECURITY_PRIVILEGES.S_SCRIPTS_WRITE, securityGroupName),
-                        }}
-                    />
+                    {
+                        console.log(actionTypes)
+                    }
+                    {
+                        matchingActionType.type === 'fwk.executeScript'
+                        && (
+                            <Box paddingX={2} className={classes.buttonContainer}>
+                                <Button
+                                    variant="contained"
+                                    color="secondary"
+                                    disabled={
+                                        !(getScriptNameOrValue(parameters, 'script').value.length
+                                            && getScriptNameOrValue(parameters, 'version').value.length)
+                                    }
+                                    size="small"
+                                    endIcon={<ChevronRightRounded />}
+                                    onClick={
+                                        actionTypes.find((element) =>
+                                            getScriptNameOrValue(parameters, 'script').value === element.type)
```

Comment:
getScriptNameOrValue(parameters, 'script') and  getScriptNameOrValue(parameters, 'version') are called a number of times. Can't you retrieve the value early on, put it in a parameter and use the parameter from that point?